docs/odh/oci-model-storage.md

```diff
@@ -0,0 +1,208 @@
+which also describes how to deploy models from OCI images. 
+
+This page offers a guide similar to the upstream project documentation, but
+focusing on the OpenDataHub and OpenShift characteristics. To demonstrate
+how to create an OCI image, the publicly available [MobileNet v2-7
+model](https://github.com/onnx/models/tree/main/validated/vision/classification/mobilenet)
+is used. This model is in ONNX format.
+
+The ODH projects provides configurations for the OpenVINO model server, which
+supports models in ONNX format. Thus, this guide will use this model server
+to demonstrate how deploy the MobileNet v2-7 model stored in an OCI image.
+
+## Storing a model in an OCI image
+
+Start by creating an empty directory for downloading the model and creating
+the necessary support files to create the OCI image. You may use a temporary
+directory by running the following command:
+```shell
+cd $(mktemp -d)
+```
+
+OpenVINO expects a specific directory tree for model versioning.
+Starting from some base directory, its contents should be a collection of
+numbered subdirectories using positive integer values. The numbers would
+represent the versions of the model. When using OCI images, this
+structure may be irrelevant, as you can use the OCI container registry
+features. However, since OpenVINO expects the versioned directory structure, a
+single subdirectory with an artificial version `1` can be used. Using `models/` as the
+base path, create the expected directory structure and download the sample
+model to it:
+```shell
+# 
+mkdir -p models/1
+
+DOWNLOAD_URL=https://github.com/onnx/models/raw/main/validated/vision/classification/mobilenet/model/mobilenetv2-7.onnx
+curl -L $DOWNLOAD_URL -O --output-dir models/1/
+```
+
+> [!TIP]
+> If you are planning to use a different model server, you should adapt this
+> guide accordingly to your model server requirements. Typically, you would need
+> to place your model files directly under `models/`.
+
+Create a file named `Containerfile` with the following contents:
+```Dockerfile
+FROM registry.access.redhat.com/ubi8/ubi-micro:latest
+COPY --chown=0:0 models /models
+RUN chmod -R a=rX /models
+```
+
+Notice that model files are copied into `/models` inside the container. KServe
+expects this path to exist in the OCI image and also expects the model files to
+be inside it.
+
+Also, notice that `ubi8-micro` is used as a base image. Empty images, like
+`scratch` cannot be used, because KServe needs to configure the model image
+with a command to keep it alive and ensure the model files remain available in
+the pod. Thus, it is required to use a base image that provides a shell.
+
+Finally, notice that ownership of the copied model files is changed to the `root`
+user and group, and also read permissions are granted to all users. This is
+important, because OpenShift runs containers with a random user ID and with the
+`root` group ID. The adjustment of the group and the privileges on the model files
```

Comment:
I am, not sure about it, avoiding user root is a good practice.
Did you try to add a dummy user and use this uid instead 0?

e.g.: `COPY --chown=1001:0 models /models`

Reply:
My understanding is that the advice against using `root` is applicable for running processes. In that case, you definitely want to avoid running as `root` because it has security implications.

In contrast, here the models are simple data files and cannot be run. The `chmod` command in the Containerfile only gives read permissions and won't give write nor execute permissions. The model server (which is another container) would load the model and run it, and would be running as non-root user (provided the user configured the `ServingRuntime` correctly).

So, given the models are simple data files, I'm not seeing security implications with giving root ownership. It  has the added benefit that the model files cannot be modified once mounted in the pod. So, I preferred to keep the Containerfile simple and not creating users, nor using dummy UIDs.

Reply:
I think I'm not understanding the relationship between bash and the data files... And the base image has a lot of files owned by root. Probably, I'm not seeing something as I don't understand the problem.

Reply:
It is more about good practices than a real problem.

Reply:
I agreed with @spolti